Update to detector docs (#608)
microsoft/component-detection: docs/detectors/npm.md

# Npm Detection

## Requirements

>     There are multiple detectors that make up npm detection and each detector searches for the following files:

Reviewer: ```suggestion
There are multiple detectors that make up npm detection and each detector searches for the following files:
```

    - [The Npm Component detector looks for `package.json`](https://github.com/microsoft/component-detection/blob/251276d7951c7eaa880ed58b1a974b25dba92cd2/src/Microsoft.ComponentDetection.Detectors/npm/NpmComponentDetector.cs#L36)
    - [The NpmLockFile detector looks for `package-lock.json`, `npm-shrinkwrap.json`,`lerna.json`](https://github.com/microsoft/component-detection/blob/251276d7951c7eaa880ed58b1a974b25dba92cd2/src/Microsoft.ComponentDetection.Detectors/npm/NpmLockfileDetectorBase.cs#L52)

## Detection strategy

>      Npm detection searches for dependencies in `packages.json`, `package-lock.json`, `npm-shrinkwrap.json`(which are autogenerated by running npm init and other commands) as well as `lerna.json` in the scan directory. The lockfile detector is able to scan for transitive dependencies within the project. There is also an extension of the lockfile detector the NpmLockFilev3 detector that is able to scan [version 3 of lockfiles](https://github.com/microsoft/component-detection/blob/251276d7951c7eaa880ed58b1a974b25dba92cd2/src/Microsoft.ComponentDetection.Detectors/npm/NpmLockfile3Detector.cs#L36)

Reviewer: ```suggestion

Npm detection searches for dependencies in `packages.json`, `package-lock.json`, `npm-shrinkwrap.json`(which are autogenerated by running npm init and other commands) as well as `lerna.json` in the scan directory. The lockfile detector is able to scan for transitive dependencies within the project. There is also an extension of the lockfile detector the NpmLockFilev3 detector that is able to scan [version 3 of lockfiles](https://github.com/microsoft/component-detection/blob/251276d7951c7eaa880ed58b1a974b25dba92cd2/src/Microsoft.ComponentDetection.Detectors/npm/NpmLockfile3Detector.cs#L36)
```

## Known limitations

> Npm has the option to download optional dependencies which can cause an overreporting issue with the detector. However this is not much of an issue currently since most projects only use dependencies and devdependencies.

Reviewer: ```suggestion

Npm has the option to download [`optionalDependencies`][1] which can cause an overreporting issue with the detector. However this is not much of an issue currently since most projects only use `dependencies` and `devDependencies`.

[1]: https://docs.npmjs.com/cli/v7/configuring-npm/package-json#optionaldependencies
```
